components/vocabularies/rest/src/test/java/org/phenotips/vocabularies/rest/internal/DefaultCategoriesResourceTest.java

```diff
@@ -0,0 +1,139 @@
+import java.util.function.Function;
+
+import javax.inject.Provider;
+import javax.ws.rs.core.UriInfo;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import com.xpn.xwiki.XWikiContext;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for the {@link DefaultCategoriesResource} class.
+ */
+public class DefaultCategoriesResourceTest
+{
+    private static final String CATEGORY_A_LABEL = "A";
+
+    private static final String CATEGORY_B_LABEL = "B";
+
+    private static final String CATEGORY_C_LABEL = "C";
+
+    @Rule
+    public MockitoComponentMockingRule<CategoriesResource> mocker =
+        new MockitoComponentMockingRule<>(DefaultCategoriesResource.class);
+
+    @Mock
+    private Provider<Autolinker> autolinkerProvider;
+
+    @Mock
+    private Category categoryA;
+
+    @Mock
+    private Category categoryB;
+
+    @Mock
+    private Category categoryC;
+
+    @Before
+    public void setUp() throws ComponentLookupException
+    {
+        MockitoAnnotations.initMocks(this);
+
+        final Execution execution = mock(Execution.class);
+        final ExecutionContext executionContext = mock(ExecutionContext.class);
+        final ComponentManager componentManager = this.mocker.getInstance(ComponentManager.class, "context");
+        when(componentManager.getInstance(Execution.class)).thenReturn(execution);
+        when(execution.getContext()).thenReturn(executionContext);
+        when(executionContext.getProperty("xwikicontext")).thenReturn(mock(XWikiContext.class));
+
+        final VocabularyManager vocabularyManager = this.mocker.getInstance(VocabularyManager.class);
+        final DomainObjectFactory objectFactory = this.mocker.getInstance(DomainObjectFactory.class);
+        final UserManager users = this.mocker.getInstance(UserManager.class);
+        final AuthorizationService authorizationService = this.mocker.getInstance(AuthorizationService.class);
+
+        final List<String> categoriesList = Arrays.asList(CATEGORY_A_LABEL, CATEGORY_B_LABEL, CATEGORY_C_LABEL);
+        when(vocabularyManager.getAvailableCategories()).thenReturn(categoriesList);
+
+        when(objectFactory.createCategoriesRepresentation(eq(categoriesList), any(Autolinker.class),
+            any(Function.class))).thenReturn(Arrays.asList(this.categoryA, this.categoryB, this.categoryC));
+
+        final User user = mock(User.class);
+        when(users.getCurrentUser()).thenReturn(user);
+        when(authorizationService.hasAccess(eq(user), eq(Right.ADMIN), any(DocumentReference.class)))
+            .thenReturn(true);
+
+        ReflectionUtils.setFieldValue(this.mocker.getComponentUnderTest(), "autolinker", this.autolinkerProvider);
```

Comment:
`ReflectionUtils` are not needed. You can directly get the autolinker as `mocker.getInstance(Autolinker.class)`, as the mocker knows how to get around a `Provider` wrapper.